### DISCUSSION TITLE
Manifest: android:screenOrientation ?

### DISCUSSION THREAD

**Discussion by dmunsie** [airsdk/Adobe-Runtime-Support#3171] [Q&A] [answered]
I'm trying to find a way to set the **screenOrientation** in the manifest, but everything is failing during the build process. Anyone know how to do this? I need to do something like this....

android:name="android:screenOrientation" android:value="userLandscape"

But nothing is working and at one point during my trials the sdk returned an error saying: "error 404: Android attribute android:screenOrientation is not allowed to be overridden". ? ? ?

Thx for any help.

**Comment by marchbold** [comment]
I think this is one of the fields that AIR takes from the values in the `initialWindow` element of your app descriptor:

https://airsdk.dev/docs/building/application-descriptor-files/elements/initialWindow#aspectratio

**Comment by dmunsie** [comment]
I believe this is different from aspect ratio. I don't know, it may need to be set via an activity?

https://developer.android.com/guide/topics/manifest/activity-element

android:[screenOrientation]=["unspecified" | "behind" |
                                     "landscape" | "portrait" |
                                     "reverseLandscape" | "reversePortrait" |
                                     "sensorLandscape" | "sensorPortrait" |
                                     "userLandscape" | "userPortrait" |
                                     "sensor" | "fullSensor" | "nosensor" |
                                     "user" | "fullUser" | "locked"]

**Reply by marchbold** [reply]
What are you trying to achieve?

**Reply by dmunsie** [reply]
Custom app for a custom designed tablet that allows device orientation changes and doesn't have a gyro. Total.. guess...guessing this is the potential issue. The other few devs on the project use Unity and they were able to make their app compatible by setting screenOrientation to "userLandscape".

**Reply by marchbold** [reply]
So to override the main air activity you need to add an activity like the following inside the `<application>` tag in your manifest additions

```xml
<activity android:screenOrientation="userLandscape">
    <intent-filter>
        <action android:name="android.intent.action.MAIN"></action>
        <category android:name="android.intent.category.LAUNCHER"></category>
    </intent-filter>
</activity>
```

But yeah AIR is flagging this as "not allowed to be overridden". @ajwfrost might need to comment here. Looks like it is normally just set to `"user"`

You could try using the android studio project output and manually overriding the value?

**Reply by dmunsie** [reply]
Thank you for taking the time to reply, much appreciated. And thank you for verifying the problem is internal with the sdk. Hopefully it's an easy update to allow that to be over written. 

"You could try using the android studio project output and manually overriding the value?"

hmm...that's beyond my knowledge of how things work, wouldn't even know where to begin. lol.. Thx.

**Reply by ajwfrost** [reply, marked as answer]
Okay so just looking at what's happening in ADT here ... we have a template manifest file which is the basis for each application, and this is basically used as a non-overridable template, except for specific attributes. The template includes `android:screenOrientation="user"` and so ADT decides that no one can override this attribute for the activity, because we don't have it on the "allow overrides" attribute list. For `activity`, the only thing on that list is `android:launchMode`...

I can't actually see why that's not allowed, and it's simple to add as into the "allow overrides" list.

If you're in a hurry, then the suggestion to build via Android Studio is probably your only option.. the simplest way to do that:
- edit a file "c:\users\username\.airsdk\adt.cfg" (or "/Users/username/.airsdk/adt.cfg") and add/comment the line `KeepAndroidStudioOutput=true`
- do your build again
- use Android Studio to open up the subfolder that would have just been created, probably alongside your app descriptor file (or basically in your working folder when you ran ADT), called "AndroidStudioProject"
- edit the AndroidManifest.xml file from within that project
- build a signed apk/bundle from within the Android Studio build menu

**Reply by dmunsie** [reply]
Thank you for the reply. Strangely enough.... when I omit the "aspect ratio" setting in the <initialWindow> settings, it works! 

I'll mark your reply as the correct answer , thank you for taking the time to reply. 

"aspectRatio - Specifies the aspect ratio of the application. Options are any, portrait or landscape.

If not specified, the application opens in the "natural" aspect ratio and orientation of the device. The natural orientation varies from device to device. Typically, it is the portrait aspect ratio on small-screen devices such as phones. On some devices, such as the iPad tablet, the application opens in the current orientation."

**Comment by dmunsie** [comment]
Also... the splash screen comes up in the CORRECT orientation, and when it goes away and the main app runs it comes up in the wrong orientation.  Strange.
